File: java/0001_TwoSum.java
/*
08DEC2025
https://leetcode.com/problems/two-sum/
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indices = new HashMap<>();

        int i = 0;
        for (int num : nums) {
            int complement = target - num;
            if (indices.containsKey(complement)) {
                return new int[] {indices.get(complement), i};
            }
            indices.put(num, i);
            i++;
        }

        return new int[]{};
        
    }
}

File: 0001_TwoSum.cs
/*
20JAN2026
1. TwoSum
https://leetcode.com/problems/two-sum/

Given an array of integers nums and an integer target, return indices
of the two numbers such that they add up to target. You may assume 
that each input would have exactly one solution, and you may not use
the same element twice. You can return the answer in any order.
*/

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var indices = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int curr = nums[i];
            int complement = target - curr;
            if (indices.ContainsKey(complement)) {
                return new int[] {indices[complement], i};
            }
            indices[curr] = i;
        }

        return new int[2];
    }
}
